R/validate.R

```diff
@@ -232,13 +232,22 @@ validate_forecast.forecast_sample <- function(data, ...) {
 #' information, messages and warnings
 #' @importFrom data.table ':=' is.data.table
 #' @importFrom checkmate assert_data_table
+#' @importFrom cli cli_abort
 #' @export
 #' @keywords internal_input_check
 validate_general <- function(data) {
   # check that data is a data.table and that the columns look fine
-  assert_data_table(data)
-  assert(check_data_columns(data))
-  data <- assure_model_column(data)
+  assert_data_table(data, min.rows = 1)
+  assert(check_columns_present(data, c("observed", "predicted", "model")))
```

Comment:
Do we ideally want a new test for this?

Reply:
We do have this existing test: 
<img width="796" alt="image" src="https://github.com/epiforecasts/scoringutils/assets/37978797/4b229384-f24f-4106-80af-08f7d6e88049">

And there is another one that captures what happens if you pass in a data.frame with 0 rows. I think these tests got introduced in #583 to address the codecov issues there